test/regexs.test.ts

```diff
@@ -119,13 +119,21 @@ const tests = {
             'a-b@example.at',
             'test@my.example.com',
             'test@my-super.example.com',
+            'a.b.c+123~@example.com',
+            'a.b.c.d@example.com',
         ],
         invalid: [
             ' test@example.com',
+            'test@example.com ',
             'test@@example.com',
             'test@localhost',
             'not an email',
             '@example.com',
+            'test..test@example.com',
+            'test.@example.com',
+            '.test@example.com',
+            '...test@example.com',
+            'test@example_example.com',
```

Comment:
from what I understood, this is a valid domain, the only place where `_` is invalid is the top level domain (so the right most part after last dot), right @fnesveda? if so would be good to at least add a comment to this one in case we decide to support this at some point.

Reply:
Uff, I have no idea anymore, the info is conflicting. Let's add a comment for now and then maybe add support in the future in a separate PR, this one is becoming too long.

Reply:
comment `no underscore in domain name now but the standards are not 100% clear, might be reviewed` added